va/va.go

```diff
@@ -173,13 +173,20 @@ func (va ValidationAuthorityImpl) getAddr(ctx context.Context, hostname string)
 	return addr, addrs, nil
 }
 
-type dialer struct {
-	record core.ValidationRecord
-	stats  metrics.Scope
+type http01Dialer struct {
+	record      core.ValidationRecord
+	stats       metrics.Scope
+	dialerCount int
 }
 
-func (d *dialer) Dial(_, _ string) (net.Conn, error) {
-	realDialer := net.Dialer{Timeout: validationTimeout}
+func (d *http01Dialer) newDialer() net.Dialer {
```

Comment:
The method naming here is a bit confusing because this is also a "dialer" struct of some sort, so I half-expected this to return a new `http01Dialer`. Maybe we can clarify that by calling this `realDialer()`?

Also, `net.Dialer` is a moderately large struct, so it's better to return a `*net.Dialer` (also the Dial() method has a pointer receiver, which Go silently casts for you, but that suggests the intended pointer vs direct usage).